plugins/inputs/slurm/slurm_test.go

```diff
@@ -0,0 +1,399 @@
+package slurm
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/influxdata/telegraf/testutil"
+	goslurm "github.com/pcolladosoto/goslurm/v0038"
+	"github.com/stretchr/testify/require"
+)
+
+func TestURLs(t *testing.T) {
+	for _, url := range []string{"http://example.com:6820", "https://example.com:6820", "http://example.com"} {
+		plugin := Slurm{
+			URL: url,
+		}
+		require.NoError(t, plugin.Init())
+	}
+
+	for _, url := range []string{
+		"httpp://example.com:6820", "httpss://example.com:6820", "", "http://:6820", "http://",
+	} {
+		plugin := Slurm{
+			URL: url,
+		}
+		require.Error(t, plugin.Init())
+	}
+}
+
+func TestPanicHandling(t *testing.T) {
+	ts := httptest.NewServer(http.NotFoundHandler())
+	defer ts.Close()
+
+	ts.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/slurm/v0.0.38/diag":
+			w.Header().Add("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+			_, err := w.Write([]byte(`{
+				"meta": {},
+				"errors": [],
+				"statistics": {
+					"rpcs_by_message_type": [],
+					"rpcs_by_user": [],
+					"jobs_running": 100
+				}
+			}`))
+			require.NoError(t, err)
+		default:
+			w.WriteHeader(http.StatusNotFound)
+			t.Fatalf("unexpected path: " + r.URL.Path)
+		}
+	})
+
+	plugin := &Slurm{
+		URL:              "http://" + ts.Listener.Addr().String(),
+		IgnoredEndpoints: []string{"jobs", "nodes", "partitions", "reservations"},
+	}
+	require.NoError(t, plugin.Init())
+
+	var acc testutil.Accumulator
+	require.NotPanics(t, func() {
+		require.NoError(t, plugin.Gather(&acc))
+	})
```

Comment:
I think this is sufficient as we will notice a panic in case it happens. ;-)
```suggestion
	require.NoError(t, plugin.Gather(&acc))
```

Reply:
Whoops! I didn't want to 'pollute' the test output with a panic 😉 I added the change on a5c910fa9171f381d048b267e3190c30bdbd0a09.